### PR TITLE
Docs: Fix design docs index page

### PR DESCRIPTION
**Pull request by joanlopez**
**What this PR does / why we need it**:

The [design docs index page](https://grafana.com/docs/loki/latest/design-documents/) is currently empty while there are some design docs available within that section. So, I thought it's good to have that index page as a shareable landing page for all the Loki design docs, which is technical valuable information.

### CODE REVIEW

File: docs/sources/design-documents/_index.md [hunk 1]
```diff
@@ -3,3 +3,8 @@ title: Design documents
 weight: 1300
 ---
 # Design documents
+
+1. [Labels from Logs](labels/)
+1. [Promtail Push API](2020-02-promtail-push-api/)
+1. [Write-Ahead Logs](2020-09-write-ahead-log/)
+1. [Ordering Constraint Removal](2021-01-ordering-constraint-removal/)
```
Comment (KMiller-Grafana) [inline docs/sources/design-documents/_index.md:10]:
```suggestion
- [Labels from Logs](labels/)
- [Promtail Push API](2020-02-promtail-push-api/)
- [Write-Ahead Logs](2020-09-write-ahead-log/)
- [Ordering Constraint Removal](2021-01-ordering-constraint-removal/)
```

**Review by KMiller-Grafana — CHANGES_REQUESTED** [review]
Changing numbering to bullets, as there is no required ordering of steps here.

**Review by KMiller-Grafana — APPROVED** [review]
thanks for the changes.

**Comment by joanlopez** [comment]
> thanks for the changes.

Thanks to you for your time on reviewing it! Feel free to merge it, although I belong to GL org, I don't seem to have merge permissions on this project.

**Pull request merged**